cmake/CommonFlags.cmake

```diff
@@ -3,20 +3,20 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL MSVC)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP /Zm200 /Zc:wchar_t-")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP /Zm200 /Zc:wchar_t-")
 
-    if(CMAKE_BUILD_TYPE STREQUAL Debug|RelWithDebInfo)
+    if(CMAKE_BUILD_TYPE MATCHES "^(Debug|RelWithDebInfo)$")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /Od /RTC1 /W3")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Od /RTC1 /W3")
     else()
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /O2 /W3")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /O2 /W3")
     endif()
-elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang|GNU)
-    if (CMAKE_SYSTEM_NAME STREQUAL Linux|Darwin)
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "^(Clang|GNU)$")
+    if (CMAKE_SYSTEM_NAME MATCHES "^(Linux|Darwin)$")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
     endif()
 
-    if(CMAKE_BUILD_TYPE STREQUAL Debug|RelWithDebInfo)
+    if(CMAKE_BUILD_TYPE MATCHES "(^Debug|RelWithDebInfo)$")
```

Comment:
I think this might be a typo?
Should be:
`if(CMAKE_BUILD_TYPE MATCHES "^(Debug|RelWithDebInfo)$")`

Reply:
It's a typo

Reply:
I'll push in a minute